src/coreclr/jit/gentree.cpp

```diff
@@ -18937,16 +18949,27 @@ GenTreeSIMD* Compiler::gtNewSIMDNode(var_types       type,
 //
 void Compiler::SetOpLclRelatedToSIMDIntrinsic(GenTree* op)
 {
-    if (op != nullptr)
+    if (op == nullptr)
     {
-        if (op->OperIsLocal())
-        {
-            setLclRelatedToSIMDIntrinsic(op);
-        }
-        else if ((op->OperGet() == GT_OBJ) && (op->AsOp()->gtOp1->OperGet() == GT_ADDR) &&
-                 op->AsOp()->gtOp1->AsOp()->gtOp1->OperIsLocal())
+        return;
+    }
+
+    if (op->OperIsLocal())
+    {
+        setLclRelatedToSIMDIntrinsic(op);
+    }
+    else if (op->OperIs(GT_OBJ))
+    {
+        GenTree* addr = op->AsIndir()->Addr();
+
+        if (addr->OperIs(GT_ADDR))
         {
-            setLclRelatedToSIMDIntrinsic(op->AsOp()->gtOp1->AsOp()->gtOp1);
+            GenTree* addrOp1 = addr->AsOp()->gtGetOp1();
+
+            if (addrOp1->OperIsLocal())
+            {
+                setLclRelatedToSIMDIntrinsic(addrOp1);
+            }
```

Comment:
It isn't clear why we were only checking for `OBJ(ADDR(LCL))` here. We also sometimes generate `BLK(ADDR(LCL))` or other nodes and so something like the following seems like it might be a better choice, given that it will cover all indirections over locals:
```
else if (op->OperIsIndir())
{
    GenTree* lcl = op->AsInidr()->Addr()->IsLocalAddrExpr();

    if (lcl != null)
    {
        setLclRelatedToSIMDIntrinsic(lcl);
    }
}
```

Reply:
We don't seem to be very consistent on what we check for calling `setLclRelatedToSIMDIntrinsic` either. The paths that create `SIMD `or `HWINTRINSIC` nodes all call `SetOpLclRelatedToSIMDIntrinsic`, while several other paths call `setLclRelatedToSIMDIntrinsic` directly.

Of those that call `setLclRelatedToSIMDIntrinsic` directly, they vary on what they check before calling `setLclRelatedToSIMDIntrinsic`. Sometimes simply checking `OperIsLocal`, sometimes checking for specific kinds of indirections, and sometimes checking all indirections. It would be nice (assuming nothing is blocking it), if all of those checks could centrally be handled here so they are consistent.